### PR TITLE
[WIP] [frameit] Ensure both texts use same baseline

### PR DESCRIPTION
**Pull request by funnel20**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
Output: 
```
Finished in 6 minutes 46 seconds (files took 3.7 seconds to load)
4675 examples, 0 failures
```
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
Output: 
```
973 files inspected, no offenses detected
```
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.
Not applicable

### Motivation and Context

This solves issue https://github.com/fastlane/fastlane/issues/10440

The current implementation applies a `.trim` to both images for Keyword and Title.
Due to typography differences (for example the "L" that needs more vertical space than an "e") the generic baseline between 2 texts is lost. This can be seen in the excellent analysis of @janpio in this comment: https://github.com/fastlane/fastlane/issues/10440#issuecomment-348010988

_Before_
iphone6-01_remotetab_framed

This solution requests the calculated trim box and offset without applying it.
By comparing the vertical top offsets of both trim boxes, the vertical offset between them can be calculated. The box with the biggest offset will be compensated to equal the same of the other, so their text baseline will be equal.
Finally a `.crop` will be applied with the (compensated) trim boxes.

_After_
iphone6-01_remotetab_framed

Test results can be found in issue comment: https://github.com/fastlane/fastlane/issues/10440#issuecomment-348760923

### Description

The changes can be found in issue comment: https://github.com/fastlane/fastlane/issues/10440#issuecomment-348760717

Please note that:

- I've no Ruby skills, so this might not be proper Production code. But the description and the code show the intended behaviour.
- All `print` commands are for debugging, so can be removed.
- Hence @janpio offered to help, so I added [WIP] to the title of the PR.
- The current code aligns the trimmed/cropped images to the top. To make this fix robust for any future changes (for example align to bottom), the height of both trim boxes (parameter 2) should be made equal: the largest after the current vertical top offset compensations is leading.

### CODE REVIEW

File: frameit/lib/frameit/editor.rb [hunk 1]
```diff
@@ -347,10 +349,60 @@ def build_title_images(max_width, max_height)
           i.interline_spacing interline_spacing if interline_spacing
           i.fill fetch_config[key.to_s]['color']
         end
-        title_image.trim # remove white space
+
+        # title_image.trim # remove white space
 
         results[key] = title_image
+
+        # Trimming of each individual image will result in the loss of the common baseline between both texts.
```
Comment (janpio) [inline frameit/lib/frameit/editor.rb]: Move this line of comment to commented out `.trim` call above, also reformulate as we only work on an individual image here (as we are in a loop) (also: can it only be 2 texts?)

File: frameit/lib/frameit/editor.rb [hunk 2]
```diff
@@ -347,10 +349,60 @@ def build_title_images(max_width, max_height)
...
+        # Hence retrieve the calculated trim bounding box, which can be used later to apply a crop on both images, while maintaining the same vertical top offset based on the smallest value from the trim box:
```
Comment (janpio) [inline frameit/lib/frameit/editor.rb]: Move stuff from "which can be used later ..." down to where this actually happens

File: frameit/lib/frameit/editor.rb [hunk 3]
```diff
@@ -347,10 +349,60 @@ def build_title_images(max_width, max_height)
...
+        calculated_trim_box = title_image.identify do |b|
+          b.format("%@") # CALCULATED: trim bounding box (without actually trimming), see: http://www.imagemagick.org/script/escape.php
+        end
+
+        # Get minimum vertical top offset from both trim boxes by converting it from string to an integer array:
```
Comment (janpio) [inline frameit/lib/frameit/editor.rb]: only working on one of them right now

File: frameit/lib/frameit/editor.rb [hunk 4]
```diff
@@ -347,10 +349,60 @@ def build_title_images(max_width, max_height)
...
+
+      # Crop images based on min_vertical_trim_offset to maintain text baseline:
+      words.each do |key|
+        # Get matching trim box:
+        trim_box = trim_boxes[key]
+        print "Trim box for key \"#{key}\" = #{trim_box}\n"
+
+        # Determine trim area:
+        if trim_box[3] > min_vertical_trim_offset
+          # The 4th parameter is the vertical top offset. When it's larger than the smallest vertical top offset, set it equal and compensate the crop height (2nd parameter):
```
Comment (janpio) [inline frameit/lib/frameit/editor.rb]: Move first part of comment to code that actually does that

File: frameit/lib/frameit/editor.rb [hunk 5]
```diff
@@ -347,10 +349,60 @@ def build_title_images(max_width, max_height)
...
+        crop_input = "#{trim_box[0]}x#{trim_box[1]}+#{trim_box[2]}+#{trim_box[3]}"
+        print "crop_input = #{crop_input}\n"
+
+        # Get text image:
+        image_to_crop = results[key]
+
+        # Crop image:
+        image_to_crop.crop(crop_input)
+
+        # Store image:
```
Comment (janpio) [inline frameit/lib/frameit/editor.rb]: Suggestion: Overwrite uncropped image with now cropped image

File: frameit/lib/frameit/editor.rb [hunk 6]
```diff
@@ -347,10 +349,60 @@ def build_title_images(max_width, max_height)
...
+        results[key] = image_to_crop
       end
+
+      # Output:
```
Comment (janpio) [inline frameit/lib/frameit/editor.rb]: output?

**Review by janpio — CHANGES_REQUESTED** [review]
Code looks pretty logical to me, maybe a bit more time investment into making the code comments more readable so someone will be able to understand all this in a few months and this should be fine!

All these `print` will have to go though, maybe they could be converted into `UI.verbose` - but I am not sure what the rules about those are and if they should be used for process output like this. Someone else will have to weigh in on that.

But besides these details: Super awesome first PR! 🥇

**Comment by janpio** [comment]
Oh, and tests are failing because of this:
```
MiniMagick::Error:
       ImageMagick/GraphicsMagick is not installed
```
Did you manually add any dependencies to your project you didn't add to the PR?
If not, any idea why this might be missing or what is using it?

**Comment by funnel20** [comment]
I have no idea about the MiniMagick error.
I didn't add anything, it's already used in this file.

The only new call I make is in row 359:
`calculated_trim_box = title_image.identify do |b|`

**Comment by funnel20** [comment]
How do you want me to respond to your code review?
Should I reply here, and if we both agree I make the changes in my local copy and push them to the PR?

**Comment by funnel20** [comment]
In order to create a robust solution, I'm eager to address my 4th note:

> The current code aligns the trimmed/cropped images to the top. To make this fix robust for any future changes (for example align to bottom), the height of both trim boxes (parameter 2) should be made equal: the largest after the current vertical top offset compensations is leading.


What do you think?

**Comment by janpio** [comment]
If you agree, just make the code changes and push them to your fork, they will be added automatically and the comment will get marked as "outdated".

If you don't agree, just respond under my comment and we can talk it out ;)

**Comment by janpio** [comment]
> I have no idea about the MiniMagick error.
> I didn't add anything, it's already used in this file.

Hmmm....

> What do you think?

To be honest that sentence is to complicated for me.

**Comment by funnel20** [comment]
This is my current solution explained:
img_4782

The 2 green boxes are the calculated trim boxes. It immediately shows the existing issue: when vertically aligning the 2 trim boxes _to their top_ (which the rest of the code does), the baseline is lost and the "e" in "eye" hovers above the "o" in mouth.
To prevent this I have extended the lowest trim box ("eye"), so their tops have the same common `min_vertical_trim_offset`.

But what happens when the code is changed to vertically align the 2 trim boxes _to the bottom_?
Then we have the issue back, only then title will be lower than the keyword.

So basically the highest trim box should be extended to match the bottom of the lowest trim box, here shown by common `max_vertical_trim_offset`:
img_4783

Now both trim boxes have the exact same height and still share their original baseline.
No matter how their are vertically aligned, the baseline will always be shared.

I hope this is more clear.

**Comment by janpio** [comment]
Ok, got that. Is there already an option to align to the bottom? (I saw something where the texts can be below the screenshots - is this it?) If not, you could probably ignore that here and just add a comment that this only workds for aligning to top.

PS: Make sure to sign the CLI and comment so the bot picks that up.

**Comment by funnel20** [comment]
There is no option or user setting. But apparently somewhere in the rest of the file, the code uses _align to top_.

I don't believe this trimming/cropping has to do with text being cut off, as the complete text is always in the trimmed image. If text is missing in the screenshot, the reason must be that the text image itself is too large or misplaced at the screenshot canvas.

I'll make a comment in the code that my solution (as well as the previous `.trim`) only works for _align to top_.

File: frameit/lib/frameit/editor.rb (hunk 6 above)
Comment (funnel20) [reply frameit/lib/frameit/editor.rb]: The last line of `def build_title_images(max_width, max_height)` is just:
`results`

I would expect:
`return results`

Similar to `def fetch_text(type)`.
But that's my lack of Ruby knowledge.

I'll remove the comment and leave the syntax untouched.

File: frameit/lib/frameit/editor.rb (hunk 5 above)
Comment (funnel20) [reply frameit/lib/frameit/editor.rb]: I'll try to combine these 3 lines:
```Ruby
# Get text image:
image_to_crop = results[key]

# Crop image:
image_to_crop.crop(crop_input)

# Store image:
results[key] = image_to_crop
```

Into 1 single:
```Ruby
# Crop image:
results[key].crop(crop_input)
```

I have no idea if this works directly on that indexed element.

Comment (funnel20) [reply frameit/lib/frameit/editor.rb]: The single line works 👍

File: frameit/lib/frameit/editor.rb (hunk 1 above)
Comment (funnel20) [reply frameit/lib/frameit/editor.rb]: Currently it are 1 (Title) or 2 (and Keyword).
Rephrased to:
# Trimming the image will result in the loss of the common baseline between the text in all images.

File: frameit/lib/frameit/editor.rb (hunk 2 above)
Comment (funnel20) [reply frameit/lib/frameit/editor.rb]: Done

File: frameit/lib/frameit/editor.rb (hunk 3 above)
Comment (funnel20) [reply frameit/lib/frameit/editor.rb]: Done

File: frameit/lib/frameit/editor.rb (hunk 4 above)
Comment (funnel20) [reply frameit/lib/frameit/editor.rb]: Done

**Comment by funnel20** [comment]
While working on the comments I just enhanced the code to implement bottom align, because that's more clear than a fuzzy TODO comment.

Here the difference is clear, while highlighting the trim box in yellow:

Original code with `.trim`:
iphone6-01_remotetab_framed trim

My current commit with `.crop` to align top:
iphone6-01_remotetab_framed vertical top

My robust solution with `.crop` for both align top and bottom:
iphone6-01_remotetab_framed vertical bottom

**Comment by funnel20** [comment]
Now I get a rubocop error that the code is 1 line too long:
```
Offenses:

frameit/lib/frameit/editor.rb:2:3: C: Class has too many lines. [321/320]
  class Editor ...
  ^^^^^^^^^^^^

973 files inspected, 1 offense detected
```

I submit it anyway, because this is a ridiculous check.

**Comment by funnel20** [comment]
I don't understand the whole Google CLA stuff, but I don't want to participate in that.
So here is my code solution, do whatever you want to do with it.

**Comment by janpio** [comment]
> I don't understand the whole Google CLA stuff, but I don't want to participate in that.
> So here is my code solution, do whatever you want to do with it.

That is unfortunately impossible as this is a Google Open Source project and only contributions with a signed CLA can be accepted.
It is pretty painless though, just fill out the form and reply here with "I signed it". 
It's basically what you wrote above ("here is the code, take it") but in legalese.

> Now I get a rubocop error that the code is 1 line too long:
> ...
> I submit it anyway, because this is a ridiculous check.

Fortunately trivial to work around:
https://stackoverflow.com/questions/40221797/rails-rubocop-disable-class-has-too-many-lines-error

> Here the difference is clear, while highlighting the trim box in yellow:

Wow, that perfectly visualizes not only why it was so shitty until now but also how you fixed it. 

Did you add the yellow with code or manually? If with code, it could be useful to leave that line/code in there commented out for future debugging or development.

**Comment by funnel20** [comment]
**Yellow highlight**
I visualised that in code by just setting the image background to yellow.
Temporary of course, so it's not committed. I think it's bad practise to leave commented debug code in Production code.
So for the record, I added these 2 lines to the existing `do`:
```Ruby
title_image.combine_options do |i|
  # Oversize as the text might be larger than the actual image. We're trimming afterwards anyway
  i.resize "#{max_width * 5.0}x#{image_height}!" # `!` says it should ignore the ratio
  i.background 'yellow'
  i.alpha 'remove'
end
```
The removal of alpha was key, see: https://github.com/minimagick/minimagick/issues/104#issuecomment-16856389

**Rubocop warning**
It seems bad practise to disable warnings. Isn't it possible to change the limit of this warning globally (i.e. to 1,000 or anything acceptable)?

**Google CLA**
To put it mildly, I'm not a fan of Google.
I have a non-Google email address as login with GitHub.
I'm not willing to create a Gmail and use that for GitHub.
So, it doesn't seem "painless".

**Comment by janpio** [comment]
> It seems bad practise to disable warnings. Isn't it possible to change the limit of this warning globally (i.e. to 1,000 or anything acceptable)?

These are set to a value the fastlane developers deem acceptable.

> I have a non-Google email address as login with GitHub.
> I'm not willing to create a Gmail and use that for GitHub.
> So, it doesn't seem "painless".

Nothing I can do about that, I am a volunteer as you and just relay the rules to you.

**Comment by funnel20** [comment]
Well, I guess you can pick up my code from here, create a PR on your account and merge it to the master?

**Comment by janpio** [comment]
No, as I don't own the copyright on your code - I didn't write it. I would break #5 of my own CLA :) Sorry.
(I pointed your "problem" out to someone from the team)

**Comment by funnel20** [comment]
I don't blame you, but what a terrible situation after a productive weekend.
Why should Google be involved in everything?
I contributed to several Githubs without this issue.

**Comment by janpio** [comment]
It is mentioned in https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md#i-want-to-contribute-to-fastlane which I didn't link to, my bad. (But maybe a link to the CONTRIBUTING file should be added to the YourFirstPR one)

>  Why should Google be involved in everything?

Because _fastlane_ is a Google Open Source project ;)

About why Google and big companies (with lots of lawyers) have CLAs:
https://opensource.google.com/docs/cla/#why

There will be a solution for this, just give it some time. For now you could make sure the tests succeed (even if it means adding a silly annotation to you don't like, it's how it works for fastlane).

**Comment by janpio** [comment]
Just sent a PR for the YourFirstPR change: https://github.com/fastlane/fastlane/pull/11103

**Review by janpio — APPROVED** [review]

**Comment by funnel20** [comment]
Thanks for the PR to change the docu.

Regarding the CLA; lawyers did their best to write tons of words in the why link.
As they are not capable in providing the answer in one or 2 simple sentences, I stopped reading because I value my time.

But consider this simple use-case:
I will continue to perform system satisfaction by creating a Gmail account like iamforcedtodothisforfastlane@gmail.com.
I sign the CLA, add this Gmail address to Github.
Then my PR turns green. Everybody (system?) happy.
After you merge it in the master, I'll delete my Gmail account and remove it from Github.

What did this system satisfaction contribute to more than that I now just say: Here is my code, you can use it?
I don't expect you to answer it, but I hope it shows the ridiculousness.

**Comment by funnel20** [comment]
I've added #11104 for the rubocop check.

**Comment by alexsinger** [comment]
@funnel20 thanks for your PR. This looks great. 

For the CLA, Google asks for it to be signed on all of their open source projects. Hopefully @janpio's PR makes it clearer for first time contributors. 

If you don't want to sign the the CLA, I can talk to our open source team about having someone else submit it on your behalf. The alternative would be for you to make an @gmail.com account (like the example you listed), and then add your GitHub email address as an alternative email on the account: https://support.google.com/accounts/answer/6316959?visit_id=1-636479951809319153-3505887489&p=emails&hl=en&rd=1#alternate and then sign the CLA. Then you don't need to make any changes to your GitHub email addresses.

**Comment by funnel20** [comment]
@alexsinger Thanks for you response.
I'm sorry to be so stubborn, but I can't believe that everyone just does what Google asks. Without even thinking why it's necessary and they are trying to accomplish with it.
As my example shows, this whole CLA seems to be a meaningless perception, as anybody can sign up for a fake Gmail account.

So to save me all these meaningless procedures (and obey as a brainless zombie to Google) I just want to hand over my code solution and let someone else commit it. It even doesn't have to be on my behalf.
Volunteering is just about helping others. Not about all this legal BS.

**Comment by alexsinger** [comment]
@funnel20 I understand your frustration with the legal stuff. I'm working with our team to make sure we can get your changes in. Thanks again for the submission and I'm sorry about the legal hurdles.

**Comment by funnel20** [comment]
@alexsinger Thanks for understanding. 

Off topic I really like to understand why these legal hurdles are introduced in a community project. In my opinion the whole Open Source / Github community was an entire force of volunteering specialists who showed the big companies that they were not alone in making quality software.

That was one of many reasons that I admire Fastlane. 
It's really bad they put an army of monitoring lawyers in our neck.

**Comment by janpio** [comment]
1. https://krausefx.com/blog/fastlane-is-now-part-of-fabric
2. https://blog.twitter.com/developer/en_us/topics/tools/2017/fabric-joins-google-developer-platform.html  
https://fabric.io/blog/fabric-joins-google  
https://krausefx.com/blog/fastlane-is-joining-google

**Comment by funnel20** [comment]
@janpio If Fastlane is so proud to be part of Google, why isn't that shown anywhere? 
There must be a reason why there isn't any Google badge or reference at:

1. The startpage https://fastlane.tools, only a tiny badge at the bottom for Fabric
2. The GitHub repo https://github.com/fastlane/fastlane
3. The Docs https://docs.fastlane.tools

**Comment by alexsinger** [comment]
@funnel20 the fastlane community and its contributors are important to us. We aim to be an open, welcoming community. Since fastlane is part of Google there a certain things we need to do, like have contributors sign the CLA. @janpio linked to it and I know it's long, but the reasons are in https://opensource.google.com/docs/cla/#why.

Submitting on your behalf has some barriers, but I'm seeing what we can do. If you'd reconsider signing the CLA so we can merge your change, it would be really helpful. Your frustration is understandable, considering the time you've put into this, and hopefully you'll consider signing the CLA so we can include your changes.

In terms of labeling Google in our docs and fastlane.tools page - we're rolling out an update soon to fastlane.tools that will make it more obvious.

**Comment by funnel20** [comment]
@alexsinger Thanks for understanding. 
As long as nobody reacts to my super simple use-case that pulverises the whole Google CLA I won't sign anything.
What is wrong with the [GitHub CLA](https://cla.github.com), which I signed for while signing up for my GitHub account with my non-Google email account?
It's super short, understandable for non-lawyers (like coders are) and covers the whole IP topic.

What's only acceptable for me is that I can register my non-Google GitHub email account at Google to sign the CLA. Instead of creating a Gmail account and being forced to use that as my primary account for GitHub.

**Comment by alexsinger** [comment]
@funnel20 I can't speak to the GitHub CLA, but the Google CLA is required and a Google account is needed to sign it. To clarify, the Google account doesn't need to be added to your GitHub account - you can include your non-Google account as alternative email https://support.google.com/accounts/answer/6316959?visit_id=1-636479951809319153-3505887489&p=emails&hl=en&rd=1#alternate and that will work.

**Comment by janpio** [comment]
> What is wrong with the GitHub CLA, ...

It belongs to the company behind Github, and only applies to projects that "belong" to that company. It doesn't apply to any other projects that are only _hosted_ on Github.

> What's only acceptable for me is that I can register my non-Google GitHub email account at Google to sign the CLA. Instead of creating a Gmail account and being forced to use that as my primary account for GitHub.

As I understood it you don't have to create a Gmail account, but a Google account, which you can do with any email address (click "I prefer to use my current email address" under the email adresse field on account creation). You then use that one to sign the CLA (and add your current Github email address to the CLA if that one is different from the one used to create the Google account) so it can connect your code commits to the signed CLA under your normal email adress (that now also has a connected Google account).

**Comment by alexsinger** [comment]
Yes, @janpio is right - you don't need a @gmail.com email address, just a Google account, and a Google account can be created using your existing email address.

**Comment by funnel20** [comment]
I signed it!

**Comment by funnel20** [comment]
I signed it!

**Comment by funnel20** [comment]
I have followed the instructions, but the bot doesn't accept it.

What have I done?

1. I logged in my existing Google account, which automatically created a Gmail address
2. It already has an alternative email address, which is the same as I use for Github.
3. When logged in with that google account, I signed the CLA. Of course it shows the primary Gmail address:
screen shot 2017-12-06 at 10 17 09
4. The last field is the Github name; so I added mine and hoped it would be OK.

As a second attempt, I have added my Gmail to my Github and verified it.
But still the bot fails. 
Please advise

**Comment by janpio** [comment]
I think your commit does not actuall _have_ an email address - or it is not connected to your Github account. So the CLA side is all good, but your commits or Github account are not ready (which @googlebot could actually check to avoid this confusion...).

Compare this: https://github.com/fastlane/fastlane/pull/11096/commits/fe366e159117d9ee5aa9f026098ceeab94d00d68 where "ohayon" has a link and profile picture, to this: https://github.com/fastlane/fastlane/pull/11096/commits/2a5789b3f52f34505ac576d5d44bb5e0c33b275c where it is just your name, unlinked, but a (?) behind it.

Use `git config --global user.email` to output your current local setting, add this to your Github account via https://github.com/settings/emails if it is not there yet. 

Maybe that fixes things.

**Comment by funnel20** [comment]
@janpio Great tip, indeed my git config has no email.
I'll add that. I suppose I have to recommit my changes in order get this email address linked to this PR.
If so, how can I do this?

**Comment by funnel20** [comment]
@googlebot You don't have to repeat this at every post. Please stay silent until I'll use the passphrase to verify the CLA again. Maybe on of your parents at Google can teach you this.

**Comment by janpio** [comment]
Ugh, good question.

I would probably just create a second branch (e.g. `frameit-vertical-text-offset2`, from master), then cherry pick the individual commits you made from the `frameit-vertical-text-offset` branch and commit them individually. Then submit a PR from the new branch and just link to this one and close it.

A bit of a mess with the discussion being here, but that should be fine.

**Comment by funnel20** [comment]
@janpio I follow you, let's see if I can master the master 🤪

**Comment by janpio** [comment]
Tests are failing because of this right now:

```
  1) Frameit Frameit::Screenshot frame! properly frame screenshots with captions that include apostrophes
     Failure/Error: Frameit::Screenshot.new('./frameit/spec/fixtures/editor/apostrophes.png', Frameit::Color::BLACK).frame!

     MiniMagick::Error:
       ImageMagick/GraphicsMagick is not installed
     # ./.bundle/ruby/2.3.0/gems/mini_magick-4.5.1/lib/mini_magick/configuration.rb:127:in `cli'
     # ./.bundle/ruby/2.3.0/gems/mini_magick-4.5.1/lib/mini_magick.rb:38:in `graphicsmagick?'
     # ./.bundle/ruby/2.3.0/gems/mini_magick-4.5.1/lib/mini_magick/tool.rb:126:in `executable'
     # ./.bundle/ruby/2.3.0/gems/mini_magick-4.5.1/lib/mini_magick/tool.rb:110:in `command'
     # ./.bundle/ruby/2.3.0/gems/mini_magick-4.5.1/lib/mini_magick/tool.rb:92:in `call'
     # ./.bundle/ruby/2.3.0/gems/mini_magick-4.5.1/lib/mini_magick/tool.rb:40:in `new'
     # ./.bundle/ruby/2.3.0/gems/mini_magick-4.5.1/lib/mini_magick/image.rb:488:in `identify'
     # ./frameit/lib/frameit/editor.rb:361:in `block in build_title_images'
     # ./frameit/lib/frameit/editor.rb:324:in `each'
     # ./frameit/lib/frameit/editor.rb:324:in `build_title_images'
     # ./frameit/lib/frameit/editor.rb:252:in `put_title_into_background'
     # ./frameit/lib/frameit/editor.rb:110:in `complex_framing'
     # ./frameit/lib/frameit/editor.rb:22:in `frame!'
     # ./frameit/lib/frameit/screenshot.rb:87:in `frame!'
     # ./frameit/spec/screenshot_spec.rb:23:in `block (4 levels) in <top (required)>'
```

Seems to me `frameit` is using `mini_magick` to provide ImageMagick:
https://github.com/fastlane/fastlane/blob/982afa330145985ae0d641de70b9c0956089c28c/fastlane.gemspec#L58
It is used here:
https://github.com/funnel20/fastlane/blob/fe366e159117d9ee5aa9f026098ceeab94d00d68/frameit/lib/frameit/editor.rb#L327
The tests fail when you try to use `.identify`, the error message says that it is missing `GraphicsMagick`.
GraphicsMagick is a fork of ImageMagick and also supported by mini_magick.

* Why does mini_magick try to use GraphicsMagick for `.identify` - does ImageMagick not have it?
* Why does this work locally for you?
* Did you manually install it?
* Is there no way to avoid installing GraphicsMagick to use `.identify`?

For reference:
https://www.imagemagick.org/script/identify.php
http://www.graphicsmagick.org/identify.html

Solution would be to add GraphicsMagick as a dependency - but _fastlane_ is very cautios with adding those. Hope we find a way we can avoid it.

**Comment by funnel20** [comment]
I have no clue why it works locally, I did not install anything intentionally.
Maybe GraphicsMagick is on my machine for another lib.

The reason why I ended up with `.identify` was that I tried to do it with `MiniMagick::Tool::Identify.new`, but couldn't find a way to let that act on the image variable `title_image` instead of a file name.
This works, when feeding an image file:
```Ruby
jpg_file = 'image.jpg'
identify = MiniMagick::Tool::Identify.new
identify.format("%@")
identify << jpg_file
id = identify.call
print "Trim result = #{id}\n"
```

The solution would be to use variable `title_image` within this code.
Any suggestion?

**Comment by funnel20** [comment]
I managed to create a new branch and commit with my private email: https://github.com/funnel20/fastlane/commit/6ba62366c1aee9211b7de3faf658a24f01ad09ca

**Comment by janpio** [comment]
That won't work (I think) as it still includes the old commits, so the CLA won't cover those again. Only the last one you added now is connected to @funnel20 :/

**Comment by funnel20** [comment]
And why are those commits still in? Can I kick them out?
I start to hate this whole endeavour

**Comment by janpio** [comment]
You have to start the branch from master - seems you did from this current branch. Then you should have the same "start state" as you had when oyu first created the branch and can get your commits you did in this branch over to the new one and commit them with the new users.

See it this way: You are learning a lot ;)

**Comment by funnel20** [comment]
Learning to never ever supply a fix for Fastlane again.
And loosing dozens of hours to spend on my work.

But I will complete this one, to stay professional.

**Comment by funnel20** [comment]
Finally committed a single commit in PR#3: #11124
The CLA is accepted 🎉

**Comment by janpio** [comment]
Awesome. Let's continue the discussion there.

**Comment by funnel20** [comment]
Can I safely delete this branch?

**Comment by janpio** [comment]
Yes.
